docs/scaling.md

```diff
@@ -0,0 +1,119 @@
+
+# Patch Details:
+
+The following are  changes present in scaling patch
+
+- Every Data node will have stateful set: currently all data nodes are part of one Statefule set, scaling needs every data node to be seperate statefulset with one replica,  every datanode resource are updated independently to corresponding statefulset, so that the control will be in the hands of operator instead of k8.
```

Comment:
Why does each data node need it's own stateful set? Don't you get the same effect by scaling up one statefulset? The reason multiple exist today is to ensure even distribution across zones, but that may be fixed in newer versions of k8s (need to test).

Reply:
If all Data nodes are part of one statefulset then individual Data nodes cannot be restarted with finer controls, for example if there are 2 data nodes in the statefulset, if the number of cpu cores or memory are updated then first data node will be restarted, after restarting the first data node, second will restarted . Before restarting the first or second data node is elastic cluster is in healthy state(green) or not will not be known. depending on the status of elastic cluster state the data node need to be restarted, this can be possible by having seperate stateful set for each data node so that status of elastic cluster will checked before restarting the data nodes one after one. If all data nodes are part one stateful set then  two things can happen, one is elastic cluster can go in to red state, second is master node can copy the shards from one data node to another data node unnecessarly.

Reply:
Pods on a stateful set have a pattern name, you can restart the pods individually if needed. There should not be a need to have multiple stateful sets. Also, we need to separate out elastic status from operations. We should be able to look up status and perform operations.

Reply:
Yes Pod are addressable, they have name pattern, suppose if we are increase the cpu from 2 to 10 for all the data nodes as part of scaling, then how to update cpu resource to 10 for each pod. the resources are tied to statefulsets.  we cannot update the resources on each pod before restarting. if we update on the stateful set then k8 will restart each pod.